js/src/lib/components/InferenceWidget/shared/WidgetWrapperV2/WidgetWrapperV2.svelte

```diff
@@ -0,0 +1,218 @@
+<script lang="ts">
+	import type { WidgetProps, ModelLoadInfo, ExampleRunOpts, WidgetInput, WidgetOutput } from "../types";
+	import type { WidgetExample, WidgetExampleAttribute } from "../WidgetExample";
+
+	type TWidgetExample = $$Generic<WidgetExample>;
+
+	import { onMount } from "svelte";
+
+	import IconCross from "../../../Icons/IconCross.svelte";
+	import WidgetInputSamples from "../WidgetInputSamples/WidgetInputSamples.svelte";
+	import WidgetInputSamplesGroup from "../WidgetInputSamplesGroup/WidgetInputSamplesGroup.svelte";
+	import WidgetFooter from "../WidgetFooter/WidgetFooter.svelte";
+	import WidgetHeader from "../WidgetHeader/WidgetHeader.svelte";
+	import WidgetInfo from "../WidgetInfo/WidgetInfo.svelte";
+	import WidgetModelLoading from "../WidgetModelLoading/WidgetModelLoading.svelte";
+	import { callInferenceApi, getModelLoadInfo, getQueryParamVal, getWidgetExample } from "../helpers";
+	import { modelLoadStates } from "../../stores";
+	import { InferenceDisplayability } from "../../../../interfaces/InferenceDisplayability";
+
+	export let apiUrl: string;
+	export let callApiOnMount: WidgetProps["callApiOnMount"];
+	export let apiToken: WidgetProps["apiToken"];
+	export let isLoading = false;
+	export let model: WidgetProps["model"];
+	export let includeCredentials: WidgetProps["includeCredentials"];
+	export let noTitle = false;
+	export let exampleQueryParams: WidgetExampleAttribute[] = [];
+
+	export let widgetInput: WidgetInput;
+	export let widgetOutput: WidgetOutput;
+	
```

Comment:
It's very unclear to me who/what sets the `WidgetInput`/`WidgetOutput` info here

Reply:
From what I understand:
- it is set by the `Input` or `Output` svelte components (which is sane - hold the validation logic close to the display logic)
- It goes all the way up the component hierarchy to `WidgetWrapper`, which uses it to filter and validate the  examples from model card

I think that:
1. the bottom-up data flow is not idiomatic and confusing.
2. Validation logic gets split between `WidgetWrapper` and the input/output components, which kinda breaks encapsulation